compiler/rustc_middle/src/ty/diagnostics.rs

```diff
@@ -142,11 +142,11 @@ pub fn suggest_arbitrary_trait_bound<'tcx>(
     if let Some((name, term)) = associated_ty {
         // FIXME: this case overlaps with code in TyCtxt::note_and_explain_type_err.
         // That should be extracted into a helper function.
-        if constraint.ends_with('>') {
-            constraint = format!("{}, {} = {}>", &constraint[..constraint.len() - 1], name, term);
+        constraint = if let Some(constraint) = constraint.strip_suffix('>') {
+            format!("{constraint}, {name} = {term}>")
         } else {
-            constraint.push_str(&format!("<{name} = {term}>"));
-        }
+            constraint + &format!("<{name} = {term}>")
```

Comment:
Can you keep the `push_str` here? I find it clearer than the use of `+` combined with reassignment.

Reply:
Yeah, sure. I did this so that I could re-use the name `constraint`, but I agree it ends up being less readable this way.